src/main/java/reactor/netty/http/server/ConnectionInfo.java

```diff
@@ -168,7 +168,7 @@ static ConnectionInfo parseXForwardedInfo(HttpRequest request, SocketChannel cha
 	private static InetSocketAddress getRemoteAddress(SocketChannel channel) {
 		if (HAProxyMessageReader.hasProxyProtocol()) {
 			InetSocketAddress remoteAddressFromProxyProtocol =
-					channel.attr(REMOTE_ADDRESS_FROM_PROXY_PROTOCOL).getAndSet(null);
+					channel.attr(REMOTE_ADDRESS_FROM_PROXY_PROTOCOL).get();
```

Comment:
Why don't we cache this in the `HttpTrafficHandler`/`Http2StreamBridgeHandler`?

Reply:
OK